savedata.rb

```diff
@@ -0,0 +1,78 @@
+# savedata.rb
+require 'json'
+
+class StoringData
+  attr_reader :app
+
+  def initialize(app)
+    @app = app
+  end
+
+  def load_data
+    load_people
+    load_books
+    load_rentals
+  end
+
+  def save_data
+    save_people
+    save_books
+    save_rentals
+  end
+
+  private
+
+  def load_people
+    people_data = JSON.parse(File.read('people.json'))
+    @app.people = people_data['people'].map { |person_data| Person.from_json(person_data, @app.books, @app.people) }
+  end
+
+  def load_books
+    books_data = JSON.parse(File.read('books.json'))
+    @app.books = books_data['books'].map { |book_data| Book.from_json(book_data) }
+  end
+
+  def load_rentals
+    rentals_json_content = File.readable?('rentals.json') ? File.read('rentals.json') : ''
+    rentals_data = parse_rentals_json(rentals_json_content)
+
+    @app.rentals = rentals_data['rentals']&.map do |rental_data|
+      Rental.from_json(rental_data, @app.books, @app.people)
+    end || []
+
+    @app.rentals.each do |rental|
+      book = Book.find_by_id(rental.book.id, @app.books)
+      book.rentals << rental
+    end
+  end
+
+  def parse_rentals_json(json_content)
+    return {} if json_content.empty?
+
+    JSON.parse(json_content)
+  rescue JSON::ParserError
+    puts "Error parsing rentals JSON: #{json_content}"
+    {}
+  end
+
+  def save_people
+    existing_people_data = File.exist?('people.json') ? JSON.parse(File.read('people.json'))['people'] : []
+    new_people_data = @app.people.map(&:to_json)
+    combined_people_data = (existing_people_data + new_people_data).uniq
+    File.write('people.json', JSON.pretty_generate(people: combined_people_data))
+  end
+
+  def save_books
+    existing_books_data = File.exist?('books.json') ? JSON.parse(File.read('books.json'))['books'] : []
+    new_books_data = @app.books.map(&:to_json)
+    combined_books_data = (existing_books_data + new_books_data).uniq
+    File.write('books.json', JSON.pretty_generate(books: combined_books_data))
+  end
```

Comment:
👏 Kudos on integrating the rental feature! 👏 However, there's a notable concern: after creating a rental and exiting with `7`, a duplicate entry for that book appears in the `books.json` file, restricting you from running the app twice. Let's work on resolving this issue to ensure a seamless experience:

- [ ] **Duplicate Entry in books.json:** I observed that exiting with `7` after a rental is causing a duplicate entry in the `books.json` file. To address this, carefully review the section of your code handling exits and ensure it appropriately manages book entries, checking for existing entries before adding a new one.

https://github.com/egichuhis/school-library-ruby/assets/55840999/5a57bf8a-a1e6-4f41-a023-d76990f32e96

By refining this aspect of your code, you'll enable smooth app runs, even after creating a rental. Keep up the excellent work! 🚀📚